currency/manager.go

```diff
@@ -539,21 +528,18 @@ func (fs FullStore) MarshalJSON() ([]byte, error) {
 	return json.Marshal(temp)
 }
 
-// copy copies and segregates pair store from internal and external calls.
-func (ps *PairStore) copy() (*PairStore, error) {
+// clone returns a deep clone of the PairStore
+func (ps *PairStore) clone() *PairStore {
 	if ps == nil {
-		return nil, errPairStoreIsNil
+		return nil
 	}
 	var assetEnabled *bool
 	if ps.AssetEnabled != nil {
 		assetEnabled = convert.BoolPtr(*ps.AssetEnabled)
 	}
 
-	enabled := make(Pairs, len(ps.Enabled))
-	copy(enabled, ps.Enabled)
-
-	avail := make(Pairs, len(ps.Available))
-	copy(avail, ps.Available)
+	enabled := slices.Clone(ps.Enabled)
+	avail := slices.Clone(ps.Available)
```

Comment:
Suggestion: can remove these declarations and use them in the Pairs Store allocation and below you can remove rFmt and cFmt and just use the methods you made.

Reply:
Fixed 30361ff1